File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap2/OpenIPC1.pl
#!/usr/bin/perl -w

### OpenIPC1.pl

use strict;
use IO::Handle;           # for autoflush()                        #(A)

print "Started OpenIPC1 process with PID $$\n";                    #(B)

open WRITE_TO_PIPE, "|cat -A |OpenIPC2.pl 2>/dev/null"             #(C)
    or die "Cannot fork: $!";                                      #(D)

local $SIG{PIPE} = sub { die "pipe has broken" };                  #(E)

WRITE_TO_PIPE->autoflush(1);                                       #(F)

while (<>) {                                                       #(G)
    print "PARENT WRITES INTO PIPELINE: $_";                       #(H)
    print WRITE_TO_PIPE;                                           #(I)
}

close WRITE_TO_PIPE;                                               #(J)

die "Non-zero exit of $?" if $?;                                   #(K)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap3/OpenIPC1.py
#!/usr/bin/python

### OpenIPC1.py

import os                         # for popen()                     #(A)
import sys                        # for stdin                       #(B)

print "Started OpenIPC1 parent process with PID", os.getpid()       #(C) 

WRITE_TO_PIPE = os.popen( "cat -A | OpenIPC2.py 2>/dev/null", 'w' ) #(D)

while ( True ):                                                     #(E)
    input = sys.stdin.readline()                                    #(F)
    WRITE_TO_PIPE.writelines( input )                               #(G)
    WRITE_TO_PIPE.flush()                                           #(H)
    print "PARENT WRITES INTO PIPELINE: ", input,                   #(I)
    if (input == ''): break                                         #(J)

WRITE_TO_PIPE.close()                                               #(K)
